packages/components/src/Select/Select.tsx

```diff
@@ -1,30 +1,125 @@
+import { HiddenSelect, useSelect } from '@react-aria/select';
+import type { AriaSelectProps } from '@react-types/select';
+import { useOverlayTriggerState } from '@react-stately/overlays';
+import { useOverlayTrigger, useOverlayPosition } from '@react-aria/overlays';
+
 import { ComponentProps } from '@marigold/types';
+import { ArrowDown, ArrowUp } from '@marigold/icons';
+import { useStyles } from '@marigold/system';
+
 import { Box } from '../Box';
+import { Label } from '../Label';
+import { ListBox } from './ListBox';
+import { Popover } from './Popover';
 
-export type SelectProps = { variant?: string } & ComponentProps<'select'>;
+export type SelectProps = {
+  placeholder?: string;
+  disabled?: boolean;
+} & ComponentProps<'select'> &
+  AriaSelectProps<object>;
 
-export const Select: React.FC<SelectProps> = ({
-  variant = 'default',
-  children,
+export const Select = ({
+  placeholder = 'Select an option',
+  disabled,
+  className,
   ...props
-}) => {
-  const iconStyles = useStyles({
-    css: {
-      alignSelf: 'center',
-      pointerEvents: 'none',
-      ml: '-28px',
-    },
+}: SelectProps) => {
+  const state = useSelectState(props);
+  const overlayTriggerState = useOverlayTriggerState({});
+  const triggerRef = useRef<HTMLElement>() as RefObject<HTMLElement>;
+  const overlayRef = useRef<HTMLDivElement>();
+  const iconClassName = useStyles({
+    css: { fill: disabled ? 'disabled' : 'text' },
+  });
+  const popoverClassName = useStyles({
+    css: { width: triggerRef.current && triggerRef.current.offsetWidth + 'px' },
   });
 
+  // Get props for the overlay
+  const { overlayProps } = useOverlayTrigger(
+    { type: 'listbox' },
+    overlayTriggerState,
+    triggerRef
+  );
+  // Get popover positioning props relative to the trigger
+  const { overlayProps: positionProps } = useOverlayPosition({
+    targetRef: triggerRef,
+    overlayRef: overlayRef as RefObject<HTMLElement>,
+    placement: 'bottom',
+    isOpen: state.isOpen,
+    onClose: state.close,
+  });
+  // Get props for child elements from useSelect
+  const { labelProps, triggerProps, valueProps, menuProps } = useSelect(
+    props,
+    state,
+    triggerRef
+  );
+  // Get props for the button based on the trigger props from useSelect
+  const { buttonProps } = useButton(triggerProps, triggerRef);
+
+  const { focusProps } = useFocusRing();
+
   return (
-    <Box display="flex">
-      <Box as="select" variant={`select.${variant}`} {...props}>
-        {children}
+    <Box position="relative" display="inline-block">
+      {props.label && (
+        <Box>
+          <Label
+            {...labelProps}
+            htmlFor={labelProps.id}
+            variant={disabled ? 'disabled' : 'above'}
+          >
+            {props.label}
+          </Label>
+        </Box>
+      )}
+      <HiddenSelect
+        state={state}
+        triggerRef={triggerRef}
+        label={props.label}
+        name={props.name}
+        isDisabled={true}
```

Comment:
Why is it disabled? shouldn't that be based on the disabled state of the whole component?

Reply:
Yes I thought give the disabled prop ist better, too. But I got this error if theres the disabled prop and I don't know how to solve it.
![image](https://user-images.githubusercontent.com/59875255/127108224-3f40465c-8227-4350-97d8-9e928a2839b1.png)

Reply:
Okay, but how are we or is react-aria using the regular select element? Why is it there if we have a complete custom select?

Reply:
I'm just trying to figure this out 🙈